### PR TITLE
Hide beatmap detail area if no beatmaps are available

### LINKED ISSUE #5300: Leaderboard sections should be hidden if no beatmap are available.
When you don't have beatmaps, `DummyWorkingBeatmap` is used as the working beatmap to tell on the wedge that there are no beatmaps, the problem is you get wrong messages when viewing the leaderboard `Leaderboards are not available for this beatmap`

the leaderboard should probably be hidden if there are no beatmaps available.

### PR DESCRIPTION
**Pull request by frenzibyte**
Closes #5300

### CODE REVIEW

File: osu.Game/Screens/Select/BeatmapDetailArea.cs [hunk 1]
```diff
@@ -29,6 +29,9 @@ public WorkingBeatmap Beatmap
                 beatmap = value;
                 Leaderboard.Beatmap = beatmap?.BeatmapInfo;
                 Details.Beatmap = beatmap?.BeatmapInfo;
+
+                // Hide detail area if no beatamps available (DummyWorkingBeatmap)
```
Comment (peppy) [inline osu.Game/Screens/Select/BeatmapDetailArea.cs]: pointless comment is pointless.

File: osu.Game/Screens/Select/BeatmapDetailArea.cs [hunk 2]
```diff
@@ -29,6 +29,9 @@ public WorkingBeatmap Beatmap
...
+                this.FadeTo(beatmap is DummyWorkingBeatmap ? 0 : 1);
```
Comment (peppy) [inline osu.Game/Screens/Select/BeatmapDetailArea.cs]: add a duration at least. let's go with 200.

Comment (iiSaLMaN) [reply osu.Game/Screens/Select/BeatmapDetailArea.cs]: yeah, we don't wanna go higher than that

**Comment by peppy** [comment]
This component wasn't really made to be used this way:

https://www.youtube.com/watch?v=eeNQUFy9LaM

**Comment by iiSaLMaN** [comment]
@peppy it should only hide if beatmaps are not available, I doubt someone will start deleting all of their beatmaps every second

**Comment by peppy** [comment]
The linked video is from filtering and a feasible scenario.

**Comment by iiSaLMaN** [comment]
> The linked video is from filtering and a feasible scenario.

still, I don't see a problem with the details being hidden, but if you still don't like this change, then this PR can be safely closed, or maybe discuss what should be hidden

**Comment by peppy** [comment]
The flashing text is an issue. It looks worse than how it looks on master. All the sub-components are accepting updates from the beatmap change but you are also hiding the parent element. It looks bad when both of these are happening.

**Comment by iiSaLMaN** [comment]
> The flashing text is an issue. It looks worse than how it looks on master. All the sub-components are accepting updates from the beatmap change but you are also hiding the parent element. It looks bad when both of these are happening.

A good solution for this might be to show the loading animation as a placeholder if the new beatmap is null, (`DummyWorkingBeatmap.BeatmapInfo` is null)

**Comment by peppy** [comment]
Test failure is likely a result of your changes - will need an update to test logic.

**Comment by iiSaLMaN** [comment]
@peppy We should be separating `No beatmaps available` Working Beatmap from DWB due to the beatmap details test using it for creating a dummy beatmap with details, maybe `NoBeatmapsAvailable : DummyWorkingBeatmap`

**Comment by peppy** [comment]
Sure. There should also be similar cases for "no matching search results" or similar, if we want to display information in relevant sections for such cases. I would do that in a separate PR though.

**Comment by peppy** [comment]
On checking with flyte, the interface should stay there and the messaging should be fixed.

**Pull request closed without merging**